### PR TITLE
zigbee: Add page describing Zigbee platform design

### PR DESCRIPTION
**Pull request by tomchy**
Add page describing the variety of Zigbee platform designs in NCS:
- Single SoC
- Single SoC with multiprotocol support
- Application-level NCP

### CODE REVIEW

**Comment by tomchy** [comment]
NCP SDK links are invalid - it is not available. This PR will be updated once NCP SDK will be released.

**Comment by wbober** [comment]
Is there a way to see a render of this?

File: doc/nrf/ug_zigbee_platform_design.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+* :ref:`ug_zigbee_platform_design_soc`
+* :ref:`ug_zigbee_platform_design_ncp`
+
+.. _ug_zigbee_platform_design_soc:
+
+System-on-Chip designs
+**********************
+
+This single-chip solution has the combined RFIC (the IEEE 802.15.4 in case of Zigbee) and processor. The ZIgbee stack and the application layer run on the local processor.
```
Comment (wbober) [inline doc/nrf/ug_zigbee_platform_design.rst:20]: typo ```ZIgbee```

File: doc/nrf/ug_zigbee_platform_design.rst [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+This design has the following advantages:
+
+* Lowest cost.
+* Lowest power consumption.
+* Lowest complexity.
+
+It also has the following disadvantages:
+
+* For some uses, the nRF528xx MCU can be too slow. For example, when the application would like to do complex data processing.
+* The application and the network share Flash and RAM space, which can limit the application functionality.
```
Comment (wbober) [inline doc/nrf/ug_zigbee_platform_design.rst:38]: ```network stack```

File: doc/nrf/ug_zigbee_platform_design.rst [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+This design has the following advantages:
+
+* Lowest cost.
+* Lowest power consumption.
+* Lowest complexity.
+
+It also has the following disadvantages:
+
+* For some uses, the nRF528xx MCU can be too slow. For example, when the application would like to do complex data processing.
```
Comment (wbober) [inline doc/nrf/ug_zigbee_platform_design.rst:37]: ```to slow to handle both application and network stack load```?

File: doc/nrf/ug_zigbee_platform_design.rst (hunk 2 above)
Comment (wbober) [reply doc/nrf/ug_zigbee_platform_design.rst:38]: ```space. This leaves less resources for the application.```?

File: doc/nrf/ug_zigbee_platform_design.rst [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+This single-chip solution has the combined RFIC (the IEEE 802.15.4 in case of Zigbee) and processor. The ZIgbee stack and the application layer run on the local processor.
+
+Single-chip, single protocol (SoC)
+==================================
+
+In this design, the application layer and the stack run on the same processor. The application uses the ZBOSS APIs directly.
+
+This is the SoC design most commonly used for end devices and routers.
+
+This design has the following advantages:
```
Comment (wbober) [inline doc/nrf/ug_zigbee_platform_design.rst:29]: The pros and cons that are listed here are relevant for both single-phy and multiprocol. I'd consider moving them into ```System-on-Chip designs``` section.

File: doc/nrf/ug_zigbee_platform_design.rst [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+For more information, see :ref:`zigbee_samples`.
+
+Single-chip, multiprotocol (SoC)
+================================
+
+With the nRF52 devices supporting multiple wireless technologies, including IEEE 802.15.4 and Bluetooth Low Energy (BLE), the application layer and the Zigbee and BLE stack run on the same processor.
+
+This design has the following advantages:
+
+* It leverages the benefits of highly integrated SoC, resulting in the lowest cost and the lowest power consumption.
```
Comment (wbober) [inline doc/nrf/ug_zigbee_platform_design.rst:53]: I'd remove this line if the comment about moving pros and cons is considered.

File: doc/nrf/ug_zigbee_platform_design.rst [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+Co-processor designs
+********************
+
+The co-processor designs can be used when a more powerful processor (host) that does not have a suitable radio interface. In these designs, the host interacts with the Zigbee network through Network Co-Processor (NCP).
+
+Split-stack co-processor design
+===============================
+
+In this design, the host processor runs only the Zigbee application layer (ZCL). The NCP runs the Zigbee MAC and the Zigbee Pro network layer as well as provides commands to execute BDB commissioning primitives. The host processor communicates with the NCP through a serial interface (typically UART).
```
Comment (wbober) [inline doc/nrf/ug_zigbee_platform_design.rst:75]: I'd replace ```typically UART``` with ```(USB or UART)```.

File: doc/nrf/ug_zigbee_platform_design.rst [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+.. figure:: /images/zigbee_platform_design_ncp.svg
+   :alt: Split Zigbee architecture.
+
+All components, relevant for NCP host applcation are available in the dedicated `NCP SDK repository on GitHub <https://github.com/nordicsemiconductor>`_:
+
+* ZBOSS NCP protocol specification.
```
Comment (wbober) [inline doc/nrf/ug_zigbee_platform_design.rst:82]: Are you sure this is going to be published?

File: doc/nrf/ug_zigbee_platform_design.rst [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+* Detailed list of supported Zigbee features.
+* Firmware files for nRF SoCs.
+* Application layer (ZCL) as well as ZBOSS NCP protocol sources.
+* Example applications and Makefiles.
+
+The NCP firmware project is available in `ZBOSS Open Initiative repository <https://bitbucket-zoi.dsr-corporation.com/projects/ZOI/repos/platform_ncs>`_ for NCS platform.
```
Comment (wbober) [inline doc/nrf/ug_zigbee_platform_design.rst:88]: I'd replace with ```The NCP firmware source is available to ZOI members <link to ZOI page>```.

**Comment by greg-fer** [comment]
> 
> 
> @greg-fer @tomchy Is there a way to see a render of this?

Hi @wbober , you can build the documentation locally. Steps are here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/doc_build.html

File: doc/nrf/ug_zigbee_platform_design.rst [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+
+This design has the following advantages:
+
+* Cost-optimized solution - uses the resources on the more powerful processor.
+* The more power consuming host processor can sleep, while the NCP device maintains connection with the Zigbee network.
+* The NCP application tipically has lower flash and RAM size requirements than a single SoC application.
```
Comment (wbober) [inline doc/nrf/ug_zigbee_platform_design.rst:94]: typo ```tipically```.

**Review by wbober — CHANGES_REQUESTED** [review]

File: doc/nrf/ug_zigbee_platform_design.rst (hunk 7 above)
Comment (tomchy) [reply doc/nrf/ug_zigbee_platform_design.rst:82]: No, I'm not. In fact - I'm not sure about any of those points and I was planning to update this list once the first SDK for NCP host will be released.

Comment (tomchy) [reply doc/nrf/ug_zigbee_platform_design.rst:82]: Reduced to pieces, that are necessary to make a release.

File: doc/nrf/ug_zigbee_platform_design.rst (hunk 8 above)
Comment (tomchy) [reply doc/nrf/ug_zigbee_platform_design.rst:88]: What kind of link should that be?
From the text - the directory within the repository, that contains NCP firmware files.

File: doc/nrf/ug_zigbee_platform_design.rst (hunk 4 above)
Comment (tomchy) [reply doc/nrf/ug_zigbee_platform_design.rst:29]: I agree. They are common. This changes a bit the scheme, but this should not be an issue.

File: doc/nrf/ug_zigbee_platform_design.rst (hunk 8 above)
Comment (wbober) [reply doc/nrf/ug_zigbee_platform_design.rst]: The aim here is to explain that you can't get the firmware unless you are a ZOI. So I'd put the proposed sentence with a link to zoi website. In the next sentence you can point to the repository as you did or remove the sentence. Up to you.

**Review by wbober — APPROVED** [review]
LGTM

File: doc/nrf/ug_zigbee_platform_design.rst (hunk 8 above)
Comment (greg-fer) [reply doc/nrf/ug_zigbee_platform_design.rst]: Link to ZOI added.

**Review by greg-fer — APPROVED** [review]
Please check and squash my edits before merge.

**Review by rlubos — APPROVED** [review]

**Pull request merged**